File: openmetadata-service/src/main/java/org/openmetadata/service/secrets/DBSecretsManager.java
/*
 *  Copyright 2021 Collate
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *  http://www.apache.org/licenses/LICENSE-2.0
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 */

package org.openmetadata.service.secrets;

import org.openmetadata.schema.security.secrets.SecretsManagerProvider;

public class DBSecretsManager extends SecretsManager {
  private static DBSecretsManager instance;

  private DBSecretsManager(SecretsManagerProvider secretsManagerProvider, SecretsConfig secretsConfig) {
    super(secretsManagerProvider, secretsConfig);
  }

  public static DBSecretsManager getInstance(SecretsConfig secretsConfig) {
    if (instance == null) {
      instance = new DBSecretsManager(SecretsManagerProvider.DB, secretsConfig);
    }
    return instance;
  }

  @Override
  protected String storeValue(String fieldName, String value, String secretId, boolean store) {
    return value;
  }

  // Nothing to delete on the Noop SM. We only delete on External SM
  @Override
  protected void deleteSecretInternal(String secretName) {}
}

File: ingestion/src/metadata/utils/secrets/db_secrets_manager.py
#  Copyright 2022 Collate
#  Licensed under the Collate Community License, Version 1.0 (the "License");
#  you may not use this file except in compliance with the License.
#  You may obtain a copy of the License at
#  https://github.com/open-metadata/OpenMetadata/blob/main/ingestion/LICENSE
#  Unless required by applicable law or agreed to in writing, software
#  distributed under the License is distributed on an "AS IS" BASIS,
#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
#  See the License for the specific language governing permissions and
#  limitations under the License.

"""
Secrets manager implementation for local secrets manager
"""
from metadata.generated.schema.security.secrets.secretsManagerProvider import (
    SecretsManagerProvider,
)
from metadata.utils.secrets.secrets_manager import SecretsManager


class DBSecretsManager(SecretsManager):
    """
    Will return the string as received. The client does not need to pick
    it up from any external resource.
    """

    provider: str = SecretsManagerProvider.db.name

    def get_string_value(self, secret_id: str) -> str:
        return secret_id
